File: packages/cortex-dart/lib/src/resources/pdf.dart
/// PDF generation resource.
library;

import '../errors.dart';
import '../http_client.dart';
import '../types.dart';

/// Provides access to the PDF generation API.
class PdfResource {
  final CortexHttpClient _client;
  final String _baseUrl;

  /// Creates a [PdfResource].
  PdfResource({
    required CortexHttpClient client,
    required String baseUrl,
  })  : _client = client,
        _baseUrl = baseUrl;

  /// Generates a PDF document.
  ///
  /// [data] contains the PDF generation parameters (content, template,
  /// options, etc.).
  Future<PdfGenerationResult> generate(Map<String, dynamic> data) async {
    if (data.isEmpty) {
      throw const CortexValidationException(
        message: 'Data must not be empty.',
        parameter: 'data',
      );
    }

    final json = await _client.post(
      '$_baseUrl/api/pdf/generate',
      body: data,
      timeout: const Duration(seconds: 120),
    );
    return PdfGenerationResult.fromJson(json);
  }
}

File: packages/cortex-js/src/resources/pdf.ts
import type { CortexClient } from '../client.js';
import type { PdfGenerateParams, PdfGenerateResponse, RequestOptions } from '../types.js';
import { ValidationError } from '../errors.js';

export class Pdf {
  private client: CortexClient;

  constructor(client: CortexClient) {
    this.client = client;
  }

  /**
   * Generate a PDF document.
   */
  async generate(params: PdfGenerateParams, options?: RequestOptions): Promise<PdfGenerateResponse> {
    if (!params.content) {
      throw new ValidationError('content is required', 'content');
    }

    const url = `${this.client._adminBaseUrl}/api/pdf/generate`;
    return this.client._request<PdfGenerateResponse>('POST', url, {
      body: params,
      requestOptions: options,
    });
  }
}
